src/game/default/g_cmd.c

```diff
@@ -757,11 +786,16 @@ _Bool G_AddClientToTeam(g_entity_t *ent, const char *team_name) {
 	if (ent->client->locals.persistent.team == team)
 		return false;
 
+	if (g_level.gameplay == GAME_DUEL && G_TeamSize(team) > 0) {
+		gi.ClientPrint(ent, PRINT_HIGH, "Only 1 player per team allowed in Duel mode\n");
+		return false;
+	}
+	
```

Comment:
It just looks tedious, in reality you just either vote or change gameplay
and it enables teams and matches automatically. We'll have to try it out
when I get back from vacation.

I do need to add some sounds to the quetoo pak though. I have them, just
need to upload.
On Jun 30, 2016 7:58 AM, "Jay Dolan" <notifications@github.com> wrote:

In src/game/default/g_cmd.c
<https://github.com/jdolan/quetoo/pull/137#discussion_r69119173>:

> @@ -757,11 +786,16 @@ _Bool G_AddClientToTeam(g_entity_t *ent, const char *team_name) {
>  	if (ent->client->locals.persistent.team == team)
>  		return false;
>
> +	if (g_level.gameplay == GAME_DUEL && G_TeamSize(team) > 0) {
> +		gi.ClientPrint(ent, PRINT_HIGH, "Only 1 player per team allowed in Duel mode\n");
> +		return false;
> +	}
> +	

I really appreciate the effort you've gone through to sanitize all of the
requisite gameplay vars to enable duel mode. I'm slightly concerned that
users might find it too tedious / difficult to setup? For this scenario, it
might be better to force all existing clients to no team affiliation when
gameplay is changed. Otherwise, enabling duel on a populated server would
require full cooperation from everyone connected. What do you think?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/jdolan/quetoo/pull/137/files/de9df12fe5a5cb1073369b0ce0c3d1dffc772747#r69119173>,
or mute the thread
<https://github.com/notifications/unsubscribe/AEB1YccM62FDJGq82AZuYsQ1PXS2O1ZSks5qQ698gaJpZM4IvCRR>
.